src/gpio.rs

```diff
@@ -279,7 +281,105 @@ where
     /// [`into_gpio_pin`]: #method.into_gpio_pin
     /// [`into_output`]: #method.into_output
     fn is_set_low(&self) -> Result<bool, Self::Error> {
-        Ok(!self.state.pin[T::PORT].read().port().bits() & T::MASK == T::MASK)
+        Ok(
+            !self.state.registers.pin[T::PORT].read().port().bits() & T::MASK
+                == T::MASK,
+        )
+    }
+}
+
+impl<'gpio, T, D> Pin<T, pin_state::Gpio<'gpio, D>>
+where
+    T: PinTrait,
+    D: direction::NotInput,
+{
+    /// Set pin direction to input
+    ///
+    /// This method is only available, if the pin is in the GPIO state and the
+    /// pin is not already in input mode, i.e. the pin direction is output or
+    /// unknown. You can enter the GPIO state using [`Pin::into_gpio_pin`].
+    ///
+    /// Consumes the pin instance and returns a new instance that is in output
+    /// mode, making the methods to set the output level available.
+    ///
+    /// # Example
+    ///
+    /// # TODO
+    /// ``` no_run
+    /// use lpc82x_hal::prelude::*;
+    /// use lpc82x_hal::Peripherals;
+    ///
+    /// let p = Peripherals::take().unwrap();
+    ///
+    /// let swm = p.SWM.split();
+    ///
+    /// // Transition pin into GPIO state, then set it to output
+    /// let mut pin = swm.pins.pio0_12
+    ///     .into_gpio_pin(&p.GPIO)
+    ///     .into_output();
+    ///
+    /// // Output level can now be controlled
+    /// pin.set_high();
+    /// pin.set_low();
+    /// ```
```

Comment:
This example is out of date. This should have been spotted by CI.

I'm fine with merging this as-is, and later fix the examples in one go. I've opened #192 to track this.